packages/bridge/src/AbsContractAdapter.ts

```diff
@@ -16,19 +17,32 @@ export abstract class AbsContractAdapter implements ContractAdapter {
   templatePath: string;
   outputPath: string;
   typeInterpreter: AbsTypeInterpreter;
-
-  constructor(protected blockchainType: string, public logDataOnly?: boolean) {}
+  protected blockchainType: string;
+  public logDataOnly?: boolean;
+  config: ContractAdapterConfig;
+
+  constructor(config?: ContractAdapterConfig) {
+    this.config = config || {};
+    this.blockchainType = this.config.blockchainType;
+    this.logDataOnly = this.config.logDataOnly;
+  }
 
   generate(outputPath: string) {
-    this.logger.debug(`output path is: ${outputPath}, blockchain type: ${this.blockchainType}`);
-
     // read template from node_modules folder
     this.templatePath = path.resolve(
```

Comment:
Could it be initialized once in constructor?

Reply:
It is a good suggestion. I would like to have a member fix it in another issue #70